src/ScenarioProcessor.ts

```diff
@@ -8,21 +8,21 @@ export class ScenarioProcessor<P extends Feature | Rule> extends PartialListProc
     private stepProcessor: StepProcessor<Scenario>;
     private tagProcessor: TagProcessor<Scenario>;
 
-    constructor(preCompiler: Partial<PreCompiler>) {
+    constructor(preCompiler?: Partial<PreCompiler>) {
         super(preCompiler);
         this.stepProcessor = new StepProcessor<Scenario>(preCompiler);
         this.tagProcessor = new TagProcessor<Scenario>(preCompiler);
     }
 
     public preFilter(e: Scenario, p: P, i: number): boolean {
-        return !this.preCompiler.preScenario || this.preCompiler.preScenario(e, p, i);
+        return !this?.preCompiler?.preScenario || this?.preCompiler?.preScenario(e, p, i);
```

Comment:
Do not use the `?.` in the first part of the expression, there must be a `preCompiler` property. Same for the 2nd part, there you only need to use it for the `preScenario`, as you call it.
Please apply the same for the rest.